alma/lib/Helpers/Admin/InsuranceHelper.php

```diff
@@ -216,4 +229,156 @@ public function saveConfigInsurance($config)
 
         $this->saveBOFormValues($config, $dbFields);
     }
+
+    /**
+     * Init Default data for Insurance.
+     * @return void
+     * @throws \PrestaShopDatabaseException
+     * @throws \PrestaShopException
+     */
+    public function initInsurance()
+    {
+        $this->installDB();
+        $this->createInsuranceProductIfNotExist();
+        $this->createAttributeGroupIfNotExist();
+    }
+
+    /**
+     * @return void
+     */
+    protected function installDB()
+    {
+        $sql = 'CREATE TABLE IF NOT EXISTS `' . _DB_PREFIX_ . 'alma_insurance_product` (
+          `id_alma_insurance_product` int(10) unsigned NOT NULL AUTO_INCREMENT,
+          `id_cart` int(10) unsigned NOT NULL,
+          `id_product` int(10) unsigned NOT NULL,
+          `id_shop` int(10) unsigned NOT NULL DEFAULT 1,
+          `id_product_attribute` int(10) unsigned NOT NULL DEFAULT 0,
+          `id_customization` int(10) unsigned NOT NULL DEFAULT 0,
+          `id_product_insurance` int(10) unsigned NOT NULL,
+          `id_product_attribute_insurance` int(10) unsigned NOT NULL,
+          `id_order` int(10) unsigned NULL,
+          `price` decimal(20,6) NOT NULL DEFAULT 0.000000,
+          `date_add` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
+           PRIMARY KEY (`id_alma_insurance_product`)
+        ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci';
+
+        \Db::getInstance()->execute($sql);
+    }
+
+    /**
+     * Create the default Insurance product
+     * @return void
+     * @throws \PrestaShopDatabaseException
+     * @throws \PrestaShopException
+     */
+    protected function createInsuranceProductIfNotExist()
+    {
+        $insuranceProduct = $this->productRepository->getProductIdByReference(
+            ConstantsHelper::ALMA_INSURANCE_PRODUCT_REFERENCE,
+            $this->context->language->id
+        );
+
+        if (!$insuranceProduct) {
+            try {
+                $product = new \Product();
```

Comment:
move the code in the try in a function create insurance inb the product repository, and in order to help you code, don't forget to  put comment like /** @var ProductCore $product **/ above line 284